challenge-manager/manager.go

```diff
@@ -167,110 +104,38 @@ func WithHeadBlockSubscriptions() Opt {
 	}
 }
 
-// New sets up a challenge manager instance provided a protocol, state manager, and additional options.
+// New sets up a challenge manager instance provided a protocol, state manager,
+// chain watcher, assertion manager, and additional options.
 func New(
-	ctx context.Context,
 	chain protocol.Protocol,
 	stateManager l2stateprovider.Provider,
-	rollupAddr common.Address,
+	watcher *watcher.Watcher,
+	assertionManager AssertionManager,
 	opts ...Opt,
 ) (*Manager, error) {
-
 	m := &Manager{
-		backend:                      chain.Backend(),
 		chain:                        chain,
 		stateManager:                 stateManager,
-		address:                      common.Address{},
+		assertionManager:             assertionManager,
+		watcher:                      watcher,
 		timeRef:                      utilTime.NewRealTimeReference(),
-		rollupAddr:                   rollupAddr,
-		chainWatcherInterval:         time.Millisecond * 500,
-		trackedEdgeIds:               threadsafe.NewMap[protocol.EdgeId, *edgetracker.Tracker](threadsafe.MapWithMetric[protocol.EdgeId, *edgetracker.Tracker]("trackedEdgeIds")),
-		batchIndexForAssertionCache:  threadsafe.NewLruMap[protocol.AssertionHash, edgetracker.AssociatedAssertionMetadata](1000, threadsafe.LruMapWithMetric[protocol.AssertionHash, edgetracker.AssociatedAssertionMetadata]("batchIndexForAssertionCache")),
+		trackedEdgeIds:               threadsafe.NewMap(threadsafe.MapWithMetric[protocol.EdgeId, *edgetracker.Tracker]("trackedEdgeIds")),
+		batchIndexForAssertionCache:  threadsafe.NewLruMap(1000, threadsafe.LruMapWithMetric[protocol.AssertionHash, l2stateprovider.AssociatedAssertionMetadata]("batchIndexForAssertionCache")),
 		notifyOnNumberOfBlocks:       1,
 		newBlockNotifier:             events.NewProducer[*gethtypes.Header](),
-		assertionPostingInterval:     time.Hour,
-		assertionScanningInterval:    time.Minute,
-		assertionConfirmingInterval:  time.Second * 10,
-		averageTimeForBlockCreation:  time.Second * 12,
 		headBlockSubscriptions:       false,
-		claimedAssertionsInChallenge: threadsafe.NewLruSet[protocol.AssertionHash](1000, threadsafe.LruSetWithMetric[protocol.AssertionHash]("claimedAssertionsInChallenge")),
+		claimedAssertionsInChallenge: threadsafe.NewLruSet(1000, threadsafe.LruSetWithMetric[protocol.AssertionHash]("claimedAssertionsInChallenge")),
```

Comment:
This 1000 that we use for our LRUs might need a bit more thinking just so that we don't have a lot of cache misses in a critical scenario. 1000 assertions is around 10.4 days worth, and we might run into surprises if all the caches miss at that point while we have a challenge ongoing

Reply:
Bumped to 1500. See previous comment about calculating this from inputs.